bundle/config/mutator/process_target_mode.go

```diff
@@ -39,7 +39,9 @@ func transformDevelopmentMode(b *bundle.Bundle) error {
 		if r.Jobs[i].Tags == nil {
 			r.Jobs[i].Tags = make(map[string]string)
 		}
-		r.Jobs[i].Tags["dev"] = b.Config.Workspace.CurrentUser.DisplayName
+		// Note: tag values in jobs must match the following pattern:
+		// ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$
+		r.Jobs[i].Tags["dev"] = b.Config.Workspace.CurrentUser.ShortName
```

Comment:
Shouldn't we just do something like https://github.com/databricks/cli/blob/99cc01c91c6f3112ab770cb612328ab17ffdc0eb/bundle/config/mutator/populate_current_user.go#L44 to make the name match this mold?

Reply:
We should, working on a fix. The allowed patterns vary per cloud so the change is a little more involved.